vaadin-testbench-integration-tests-junit5/src/test/java/com/vaadin/tests/ElementQueryIT.java

```diff
@@ -305,4 +307,163 @@ void allWithoutLightTheme() {
+    }
+
+    @BrowserTest
+    public void labelAndPlaceholderMatches() {
+        openTestURL();
+        TemplateViewElement view = $(TemplateViewElement.class).waitForFirst();
+
+        List<LabelPlaceholderElement> labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withLabel("one")
+                .withPlaceholder("two")
+                .all();
+        Assertions.assertEquals(1, labelPlaceholderElements.size());
+
+        labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withLabel("two")
+                .withPlaceholder("one")
+                .all();
+        Assertions.assertEquals(1, labelPlaceholderElements.size());
+    }
+
+    @BrowserTest
+    public void labelAndPlaceholderContains() {
+        openTestURL();
+        TemplateViewElement view = $(TemplateViewElement.class).waitForFirst();
+
+        List<LabelPlaceholderElement> labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withLabelContaining("n")
+                .withPlaceholderContaining("w")
+                .all();
+        Assertions.assertEquals(1, labelPlaceholderElements.size());
+
+        labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withLabelContaining("o")
+                .withPlaceholderContaining("o")
+                .all();
+        Assertions.assertEquals(2, labelPlaceholderElements.size());
+    }
+
+    @BrowserTest
+    public void captionMatches() {
+        openTestURL();
+        TemplateViewElement view = $(TemplateViewElement.class).waitForFirst();
+
+        List<LabelPlaceholderElement> labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withCaption("one")
+                .all();
+        Assertions.assertEquals(3, labelPlaceholderElements.size());
+
+        labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withCaption("two")
+                .all();
+        Assertions.assertEquals(3, labelPlaceholderElements.size());
+
+        List<NativeButtonElement> nativeButtonElements = view.$(NativeButtonElement.class)
+                .withCaption("Button with special id")
+                .all();
+        Assertions.assertEquals(1, nativeButtonElements.size());
+    }
+
+    @BrowserTest
+    public void captionContains() {
+        openTestURL();
+        TemplateViewElement view = $(TemplateViewElement.class).waitForFirst();
+
+        List<LabelPlaceholderElement> labelPlaceholderElements = view.$(LabelPlaceholderElement.class)
+                .withCaptionContaining("Special")
+                .all();
+        Assertions.assertEquals(2, labelPlaceholderElements.size());
+
+        List<NativeButtonElement> nativeButtonElements = view.$(NativeButtonElement.class)
+                .withCaptionContaining("Special")
+                .all();
+        Assertions.assertEquals(2, nativeButtonElements.size());
```

Comment:
Same as above, maybe better having different size for the two searches, plus adding a "no elements found" case